src/kernelci_pipeline/logspec_api.py

```diff
@@ -188,15 +179,10 @@ def generate_issues_and_incidents(result_id, log_url, object_type, oo_client):
     incidents = []
     for error in error_list:
         if error and error['error'].get('signature'):
-            issue = get_issue_from_db(oo_client, error['error']['signature'])
-            if issue:
-                issue_id = issue.id
-                issue_version = issue.version_num
-            else:
-                issue = new_issue(error, object_type)
-                issues.append(issue)
-                issue_id = issue["id"]
-                issue_version = issue["version"]
+            issue = new_issue(error, object_type)
+            issues.append(issue)
```

Comment:
Could it happen that you would submit an array containing duplicate issues to KCIDB here? Not a correctness problem, but would be a waste of resources.

Reply:
yes, good catch, I just added another commit to this PR that remove duplicated issues, please check